### PR TITLE
feat: add signalized failure pack builder

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Adds a signalized-crossing failure-pack builder for issue #2754 plus focused tests and a tracked smoke evidence pack.

## Linked Issues
- Closes `#2754`

## Stack / Dependency
- Base dependency: none
- Required prior PRs: none
- Stack follow-up issues: none
- Safe to review independently: yes
- Review dependency reason, if any: NA

## What Changed
- Added `scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py`.
- Added tests for real failure output, negative-control output, proxy/unavailable downgrade behavior, and required field extraction.
- Added first-use smoke evidence under `docs/context/evidence/issue_2754_signalized_failure_pack_smoke/`.

## Why It Matters
- Added value: issue #2754 now has a concrete, reproducible pack builder instead of only a requested analysis shape.
- Expected impact: reviewers can inspect compact failure-pack JSON that ties a trace fixture to denominator status and claim wording.
- Why this is worth merging now: it creates the analysis surface needed to review signalized-crossing failures without copying raw traces into git.

## Research Result Guidance
- Target claim / hypothesis / blocker this should affect: signalized-crossing failure cases can be packed with trace range, signal phase, stop-line geometry, metric row, denominator status, artifact status, and claim wording.
- Comparator or baseline, if applicable: previous state had no #2754-specific pack builder.
- Evidence tier: targeted smoke.
- Result classification: diagnostic-only / blocker-resolution.
- Decision or stop rule, if applicable: smoke fixture must produce the required pack fields and avoid benchmark/compliance claims.
- Parent issue, claim map, registry, context note, or synthesis surface to update: issue #2754.
- New research/benchmark/metric/paper-facing analysis tool, if any: representative use on durable/versioned input is tracked in `docs/context/evidence/issue_2754_signalized_failure_pack_smoke/summary.json`.

## Falsification / Non-Transfer Check
- Did the mechanism activate? yes, on the tracked signalized trace fixture plus synthetic metric row.
- Did the intervention change command source, selected command, trajectory, or route progress? NA; analysis tool only.
- Did the scenario actually contain the targeted failure mode? the smoke metric row is synthetic and intentionally marked as smoke-only, not benchmark evidence.
- Result route: continue.
- Follow-up question or issue for weak, negative, or non-transfer results: none opened; future real-trace packs can reuse the script when durable runtime rows are available.

## Next Empirical Action
- Rerun needed: no for smoke proof; yes before any benchmark or figure claim.
- Extractor or analysis tool needed: no for this scoped issue.
- Artifact missing or unavailable: no durable real-failure benchmark trace is claimed here.
- Stop / revise / continue decision: continue for smoke tooling; do not promote to benchmark evidence.
- Proposed child issue or existing follow-up: NA.

## Validation / Proof
- Commands run:
  - `uv run python scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py --help`
  - `uv run pytest tests/analysis/test_build_signalized_crossing_failure_pack_issue_2754.py -q`
  - `uv run ruff check scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py tests/analysis/test_build_signalized_crossing_failure_pack_issue_2754.py`
  - `uv run ruff format --check scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py tests/analysis/test_build_signalized_crossing_failure_pack_issue_2754.py`
  - `git diff --check -- scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py tests/analysis/test_build_signalized_crossing_failure_pack_issue_2754.py`
  - `uv run python scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py --traces tests/fixtures/analysis_workbench/simulation_trace_export_v1/issue_2868_signalized_crossing_fixture_0000.json --episodes-jsonl docs/context/evidence/issue_2754_signalized_failure_pack_smoke/episodes.jsonl --artifact-status current --allowed-claim-wording "Smoke fixture evidence only: this pack validates signalized failure-pack extraction fields against a tracked trace fixture and synthetic metric row; it is not benchmark evidence, dissertation figure evidence, or traffic-light compliance evidence." --output-json docs/context/evidence/issue_2754_signalized_failure_pack_smoke/summary.json`
- Evidence that the change works here: focused tests pass and tracked `summary.json` contains one smoke fixture case with the required pack fields.
- Benchmarks or smoke tests, if applicable: smoke fixture evidence only.

Full readiness caveat:
- `BASE_REF=origin/main scripts/dev/pr_ready_check.sh` was run through `scripts/dev/run_compact_validation.py` and failed because unrelated test collection imports `torch`, which is absent in this local worktree environment. The same run also reports no PR body source because it was executed before PR creation.

## Risks / Rollout
- Compatibility risks: input metric-row shapes may vary; the script accepts JSONL records with metrics and optional scenario metadata.
- Failure modes: real traces with different signal metadata placement may need an adapter; unavailable/proxy signal rows are intentionally downgraded.
- Rollback or fallback plan: remove the standalone script and evidence directory.

## Docs / Provenance
- Updated docs: `docs/context/evidence/issue_2754_signalized_failure_pack_smoke/README.md`.
- Relevant design or provenance notes: smoke evidence uses a tracked trace fixture and a compact synthetic metric row; no raw trace duplication.
- Any assumptions that need to be preserved: this PR does not establish benchmark, dissertation figure, or traffic-light compliance evidence.

## Downstream Propagation
- Parent issue updated (yes/no/NA): yes, via PR link and close keyword.
- Claim map / benchmark report updated (yes/no/NA): NA.
- Leaderboard / artifact catalog updated (yes/no/NA): NA.
- Registry or config index updated (yes/no/NA): NA.
- Context index / memory note updated (yes/no/NA): NA; evidence is colocated under `docs/context/evidence/`.
- Follow-up issue opened for deferred propagation (yes/no/NA): NA.
- Not applicable because: this is a scoped smoke analysis tool and evidence pack, not a benchmark result.

## Follow-Up Issues
- Deferred work: use the builder on real durable signalized failure traces before any figure or benchmark recommendation.
- Issues opened for follow-up: none.

## Reviewer Notes
- Anything a reviewer should verify closely: claim wording and denominator downgrade behavior.
- Any known limitations: smoke evidence uses a synthetic metric row; it is not benchmark evidence.

## Summary by CodeRabbit

* **Documentation**
  * Added comprehensive evidence documentation and fixtures for signalized crossing failure pack validation and analysis.

* **Tests**
  * Expanded test coverage for failure pack generation across success, failure, and edge-case scenarios.

### CODE REVIEW

File: docs/context/evidence/issue_2754_signalized_failure_pack_smoke/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Reproduction
+
+```bash
+uv run python scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py \
+  --traces tests/fixtures/analysis_workbench/simulation_trace_export_v1/issue_2868_signalized_crossing_fixture_0000.json \
+  --episodes-jsonl docs/context/evidence/issue_2754_signalized_failure_pack_smoke/episodes.jsonl \
+  --artifact-status current \
+  --allowed-claim-wording "Smoke fixture evidence only: this pack validates signalized failure-pack extraction fields against a tracked trace fixture and synthetic metric row; it is not benchmark evidence, dissertation figure evidence, or traffic-light compliance evidence." \
+  --output-json docs/context/evidence/issue_2754_signalized_failure_pack_smoke/summary.json
+```
```
[the remark this answers is not in the document]
Comment (ll7) [reply docs/context/evidence/issue_2754_signalized_failure_pack_smoke/README.md:38]: Fixed in 7e4f37b81: added an Execution Metadata section documenting the trace planner id, fixture/synthetic-row execution mode, and analysis-only smoke boundary.

File: scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+def load_trace_with_fallback(path: Path) -> SimulationTraceExport:
+    """Load a simulation trace export, falling back to a raw JSON load if validation fails."""
+    try:
+        return load_simulation_trace_export(path)
+    except Exception:
+        # Fall back to raw JSON load and bypass strict schema checks
+        from robot_sf.analysis_workbench.simulation_trace_export import (
+            SimulationTraceFrame,
+            SimulationTraceSource,
+        )
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/analysis/build_signalized_crossing_failure_pack_issue_2754.py]: Fixed in 7e4f37b81: narrowed the fallback to SimulationTraceExportValidationError so the raw loader only handles strict trace-schema validation failures.
